packages/vite/src/node/optimizer/optimizer.ts

```diff
@@ -228,6 +250,13 @@ async function createDepsOptimizer(
             const deps = await discover.result
             discover = undefined
 
+            const manuallyIncluded = Object.keys(manuallyIncludedDepsInfo)
+            discoveredDepsWhileScanning.push(
+              ...Object.keys(metadata.discovered).filter(
+                (dep) => !deps[dep] && !manuallyIncluded.includes(dep),
+              ),
+            )
```

Comment:
In case we issue the warning to add dependencies to `optimize.include`, I think we should also ask the user to add the deps discovered while the scanner was running. The scanner may end before some of these are found during the crawling of static imports in future cold starts.

Reply:
IIUC you mean we also log the dependencies found by the scanner? (Not the static imports crawler) I'm not sure if it would be ideal to request adding them to `optimizeDeps.include` manually. I think it would be enough to request from the crawler only.

In general, I think missed dependencies could mean a bug in our code (e.g. `new URL('./somewhere', import.meta.url)` which I never got around fixing 😅 ), or a `optimizeDeps.entries` misconfiguration. So we can be a bit conservative in requesting adding to `optimizeDeps.include` and try to fix the root cause instead.

Reply:
Ah ok thanks for the explanation. I don't have a preference for the names, but whichever you think fits better here.